File: app/src/main/java/vergecurrency/vergewallet/view/base/BaseActivity.java
package vergecurrency.vergewallet.view.base;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import vergecurrency.vergewallet.helpers.utils.LanguagesUtils;
import vergecurrency.vergewallet.helpers.utils.ThemeUtils;
import vergecurrency.vergewallet.service.model.PreferencesManager;

public class BaseActivity extends AppCompatActivity {

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		ThemeUtils.setTheme(PreferencesManager.getCurrentTheme(),this,false);
		super.onCreate(savedInstanceState);
	}

	@Override
	protected void attachBaseContext(Context base) {

		super.attachBaseContext(updateBaseContextLocale(base));
	}

	private Context updateBaseContextLocale(Context context) {
		return LanguagesUtils.setLocale(context, PreferencesManager.getCurrentLanguage());
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/view/base/BaseActivity.kt
package vergecurrency.vergewallet.view.base

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import vergecurrency.vergewallet.helpers.utils.LanguagesUtils
import vergecurrency.vergewallet.helpers.utils.UIUtils
import vergecurrency.vergewallet.service.model.PreferencesManager

open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        UIUtils.setTheme(PreferencesManager.currentTheme!!, this, false)
        super.onCreate(savedInstanceState)
    }

    override fun attachBaseContext(base: Context) {

        super.attachBaseContext(updateBaseContextLocale(base))
    }

    private fun updateBaseContextLocale(context: Context): Context {
        return LanguagesUtils.setLocale(context, PreferencesManager.currentLanguage!!)
    }
}
